gtsfm/bundle/bundle_adjustment.py

```diff
@@ -58,71 +170,47 @@ def run(self, initial_data: GtsfmData) -> Tuple[GtsfmData, GtsfmMetricsGroup]:
             return initial_data
 
         # noise model for measurements -- one pixel in u and v
-        measurement_noise = gtsam.noiseModel.Isotropic.Sigma(IMG_MEASUREMENT_DIM, 1.0)
+        measurement_noise = Isotropic.Sigma(IMG_MEASUREMENT_DIM, MEASUREMENT_NOISE_SIGMA)
 
         # Create a factor graph
-        graph = gtsam.NonlinearFactorGraph()
-
-        # Add measurements to the factor graph
-        for j in range(initial_data.number_tracks()):
-            track = initial_data.get_track(j)  # SfmTrack
-            # retrieve the SfmMeasurement objects
-            for m_idx in range(track.number_measurements()):
-                # i represents the camera index, and uv is the 2d measurement
-                i, uv = track.measurement(m_idx)
-                # note use of shorthand symbols C and P
-                graph.add(GeneralSFMFactorCal3Bundler(uv, measurement_noise, C(i), P(j)))
-
-        # get all the valid camera indices, which need to be added to the graph.
-        valid_camera_indices = initial_data.get_valid_camera_indices()
-
-        # Add a prior on first pose. This indirectly specifies where the origin is.
-        graph.push_back(
-            gtsam.PriorFactorPinholeCameraCal3Bundler(
-                C(valid_camera_indices[0]),
-                initial_data.get_camera(valid_camera_indices[0]),
-                gtsam.noiseModel.Isotropic.Sigma(PINHOLE_CAM_CAL3BUNDLER_DOF, 0.1),
-            )
-        )
-        # Also add a prior on the position of the first landmark to fix the scale
-        graph.push_back(
-            gtsam.PriorFactorPoint3(
-                P(0), initial_data.get_track(0).point3(), gtsam.noiseModel.Isotropic.Sigma(POINT3_DOF, 0.1)
-            )
-        )
+        graph = NonlinearFactorGraph()
 
         # Create initial estimate
-        initial = gtsam.Values()
+        initial_values = Values()
 
-        # add each PinholeCameraCal3Bundler
-        for i in valid_camera_indices:
-            camera = initial_data.get_camera(i)
-            initial.insert(C(i), camera)
-
-        # add each SfmTrack
+        # adding factors for measurements and 3D points's initial values
         for j in range(initial_data.number_tracks()):
             track = initial_data.get_track(j)
-            initial.insert(P(j), track.point3())
+            self.__add_measurement_factor(graph, track, j, measurement_noise)
+            initial_values.insert(P(j), track.point3())
+
+        # add prior and initial values for cameras
+        for i in initial_data.get_valid_camera_indices():
+            initialized_cam = initial_data.get_camera(i)
+
+            self.__add_camera_prior(graph, initialized_cam, i)
+            self.__add_camera_initial_value(initial_values, initialized_cam, i)
```

Comment:
i actually find this helper a bit less readable than before. since basically we are wrapping a 1-liner. and i think the 1-liner is clearer to me than the new method